### ISSUE TITLE
OCSP responder in openssl tool refuses to process requests from s_server

### ISSUE THREAD

**Issue by tomato42** [openssl/openssl#15002]
When using master branch (1f3b58d8413cf) the `openssl ocsp` refuses to process OCSP queries produced by `openssl s_server`

Reproducer:
```
wget https://raw.githubusercontent.com/redhat-qe-security/certgen/master/certgen/lib.sh
source lib.sh
x509KeyGen ca
x509KeyGen server
x509KeyGen ocsp

x509SelfSign ca
x509CertSign --CA ca server
x509CertSign --CA ca ocsp --basicKeyUsage critical,digitalSignature,nonRepudiation --extendedKeyUsage critical,ocspSigning --ocspNoCheck --DN CN=ocspserver


openssl ocsp -index ca/index.txt -port 8888 -rsigner $(x509Cert ocsp) -rkey $(x509Key ocsp) -CA $(x509Cert ca) >ocsp.log 2>ocsp.err &
ocsp_pid=$!


openssl s_server -www -CAfile $(x509Cert ca) -status -status_url http://localhost:8888 -key $(x509Key server) -cert $(x509Cert server) > server.log 2> server.err &
server_pid=$!

openssl s_client -CAfile $(x509Cert ca) -verify_return_error -verify 2 -connect localhost:4433 -status < /dev/null
```

Output from `s_client`:
```
OCSP response: 
======================================
OCSP Response Data:
    OCSP Response Status: malformedrequest (0x1)
======================================
```

Output from ocsp (`ocsp.log`):
```
Responder Error: malformedrequest (1)
```
Output from ocsp (`ocsp.err`):
```
openssl ocsp: waiting for OCSP client connections...
openssl ocsp: Error parsing request
400788D5AC7F0000:error:068000A8:asn1 encoding routines:asn1_check_tlen:wrong tag:crypto/asn1/tasn_dec.c:1143:
400788D5AC7F0000:error:0688010A:asn1 encoding routines:asn1_d2i_ex_primitive:nested asn1 error:crypto/asn1/tasn_dec.c:708:
400788D5AC7F0000:error:0688010A:asn1 encoding routines:asn1_template_noexp_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:641:Field=responseStatus, Type=OCSP_RESPONSE
```

**Comment by t8m** [comment]
Are you absolutely sure that this happens with current master? I might be confused but IMO this was supposed to be fixed by ef7ae359109c7c4edb252523d5ab1467ae1d2110

**Comment by tomato42** [comment]
did a clean compile and can't reproduce it with 1f3b58d, must have mixed up commits or compiles, so looks like it was fixed with ef7ae35

closing as duplicate of #13904
